diff --git a/mytube.module b/mytube.module
index 3c7393e..39ed78a 100644
--- a/mytube.module
+++ b/mytube.module
@@ -719,10 +719,30 @@ function _mytube_download_thumb($url) {
 function _mytube_ytid($url_input) {
   // FIXME: This below regex should also account for youtube-nocookie, instead of relying on this other function to swap it out
   $url = str_replace("youtube-nocookie", "youtube", $url_input);
-  $rex = "{youtube.com/v/([^\?|\&|\s]*)}si";
-  preg_match($rex, $url, $matches);
-  $ytid = $matches[1];
-  return $ytid;
+
+  // @see media_youtube/includes/MediaInternetYouTubeHandler.inc
+  //      MediaInternetYouTubeHandler::parse()
+  // http://youtube.com/watch/*
+  // http://youtube.com/embed/*
+  // http://youtube.com/v/*
+  // http://youtube.com/?v=*
+  // http://youtu.be/*
+  // http://gdata.youtube.com/feeds/api/videos/*
+  $patterns = array(
+    '@youtube\.com/watch[#\?].*?v=([^"\& ]+)@i',
+    '@youtube\.com/embed/([^"\&\? ]+)@i',
+    '@youtube\.com/v/([^"\&\? ]+)@i',
+    '@youtube\.com/\?v=([^"\& ]+)@i',
+    '@youtu\.be/([^"\&\? ]+)@i',
+    '@gdata\.youtube\.com/feeds/api/videos/([^"\&\? ]+)@i',
+  );
+
+  foreach ($patterns as $pattern) {
+    preg_match($pattern, $url_input, $matches);
+    if (isset($matches[1])) {
+      return $matches[1];
+    }
+  }
 }
 
 /**
